component/mimir/rules/types.go

```diff
@@ -0,0 +1,60 @@
+package rules
+
+import (
+	"time"
+
+	"github.com/grafana/agent/pkg/flow/rivertypes"
+)
+
+type Arguments struct {
+	ClientParams         ClientArguments `river:"client,block"`
+	SyncInterval         time.Duration   `river:"sync_interval,attr,optional"`
+	MimirNameSpacePrefix string          `river:"mimir_namespace_prefix,attr,optional"`
+
+	RuleSelector          LabelSelector `river:"rule_selector,block,optional"`
+	RuleNamespaceSelector LabelSelector `river:"rule_namespace_selector,block,optional"`
+}
+
+func setDefaultArguments(args *Arguments) {
+	if args.SyncInterval == 0 {
+		args.SyncInterval = 30 * time.Second
+	}
+
+	if args.MimirNameSpacePrefix == "" {
+		args.MimirNameSpacePrefix = "agent"
+	}
+}
+
+type LabelSelector struct {
+	MatchLabels      map[string]string `river:"match_labels,attr,optional"`
+	MatchExpressions []MatchExpression `river:"match_expressions,attr,optional"`
```

Comment:
Is there value in having this be an attribute instead of a block? Blocks are typically the default way of grouping settings together. Object attributes are desirable only if you anticipate someone loading an entire `match_expression` value from another component/expression (such as `json_decode`).